omegaconf/basecontainer.py

```diff
@@ -123,8 +123,9 @@ def __setstate__(self, d: Dict[str, Any]) -> None:
     def __delitem__(self, key: Any) -> None:
         ...
 
+    @abstractmethod
     def __len__(self) -> int:
-        return self.__dict__["_content"].__len__()  # type: ignore
+        ...
```

Comment:
It seems that you ended up implementing the same method in both subclasses (up to the assert on type) => wouldn't it actually make sense to keep its implementation in BaseContainer?

Reply:
I had a comment pending explaining it exactly on this line.